src/utils/messageParser.ts

```diff
@@ -13,14 +14,33 @@ export function parseMessage(message: string): ParsedMessage {
     }
 }
 
-function extractURLFromMessage(message: string) {
+/**
+ * Checks if a URL corresponds to a recognized social media platform.
+ * @param {string} urlStr - The URL to check.
+ * @returns {boolean} True if the URL is a recognized social media platform, false otherwise.
+ */
+function isSocialMediaUrl(urlStr:string): boolean {
+    return SOCIAL_MEDIA_URLS.some(hosts=>hosts.includes(urlStr))
+}
+
+/**
+ * Extracts a URL from a given message using a regular expression.
+ * @param {string} message - The message from which to extract the URL.
+ * @throws {Error} If a URL cannot be extracted from the message or if the URL is not a recognized social media platform.
+ * @returns {string} The extracted URL.
+ */
+function extractURLFromMessage(message: string): string {
     const urlRegex = /https?:\/\/\S+\b/;
 
     const match = message.match(urlRegex);
     if(!match){
         throw new Error('Failed to extract URL from message');
     }
-    return match[0];
+    const [urlStr]=match
```

Comment:
sometimes `match` has more than one items in its array, so it's better to always pick up the first element
```suggestion
    const urlStr=match[0]
```